native/src/routes/__tests__/TuNews.spec.tsx

```diff
@@ -38,78 +48,77 @@ const news: [TunewsModel, TunewsModel] = [
-    const cityModel = new CityModel({
-      name: 'Oldtown',
-      code: 'oldtown',
-      live: false,
-      eventsEnabled: true,
-      offersEnabled: true,
-      poisEnabled: false,
-      localNewsEnabled: true,
-      tunewsEnabled: true,
-      sortingName: 'Oldtown',
-      prefix: 'GoT',
-      latitude: 48.369696,
-      longitude: 10.892578,
-      aliases: null,
-      boundingBox: null,
-    })
-    const props = { cityModel, language, selectNews, changeUnavailableLanguage }
-    return render(
+  const navigation = createNavigationScreenPropMock<NewsRouteType>()
+  const route = {
+    key: 'route-id-0',
+    params: {
+      newsType: TU_NEWS_TYPE,
+      newsId: null,
+    },
+    name: NEWS_ROUTE,
+  }
+
+  const cities = new CityModelBuilder(3).build()
+  const city = cities[0]!
+  const languages = new LanguageModelBuilder(3).build()
+  const language = languages[0]!
+
+  const data = {
+    cities,
+    languages,
+    city,
+    language,
+    categories: new CategoriesMapModelBuilder(city.code, language.code).build(),
+    events: [],
+    pois: [],
+  }
+
+  const renderNews = () =>
+    render(
       <ThemeProvider theme={buildConfig().lightTheme}>
-        <TuNews {...props} newsId={newsId} />
+        <TuNews data={data} route={route} navigation={navigation} selectNews={selectNews} />
       </ThemeProvider>
     )
-  }
   const elementResponse = { data: news[0], error: null, loading: false, refresh: jest.fn() }
-  const tuNewsResponse = { ...elementResponse, loadMore, loadingMore: false, refresh, data: news, availableLanguages }
+  const tuNewsResponse = {
+    ...elementResponse,
+    loadMore,
+    loadingMore: false,
+    refresh,
+    data: news,
+    availableLanguages: languages,
+  }
 
   it('should show news list', () => {
-    mocked(useLoadFromEndpoint).mockImplementation(() => elementResponse)
     mocked(useLoadTuNews).mockImplementation(() => tuNewsResponse)
 
-    const { getByText } = renderNews({})
+    const { getByText } = renderNews()
     expect(getByText(news[0].title)).toBeTruthy()
     expect(getByText(news[1].title)).toBeTruthy()
 
     fireEvent.press(getByText(news[1].title))
     expect(selectNews).toHaveBeenCalledWith(news[1].id.toString())
   })
 
-  it('should show language selector if language not available', () => {
-    mocked(useLoadFromEndpoint).mockImplementation(() => elementResponse)
-    mocked(useLoadTuNews).mockImplementation(() => tuNewsResponse)
+  it('should show language selector if language not available', async () => {
+    mocked(useLoadTuNews).mockImplementation(() => ({
+      ...tuNewsResponse,
+      availableLanguages: languages.slice(1),
+    }))
 
-    const { getByText, queryByText } = renderNews({ language: 'es' })
-    expect(getByText('languageNotAvailable')).toBeTruthy()
+    const { getByText, queryByText } = renderNews()
+    await waitFor(() => expect(getByText('languageNotAvailable')).toBeTruthy())
```

Comment:
findByText already wraps waitFor internally and should therefore also work here
```suggestion
    expect(await findByText('languageNotAvailable')).toBeTruthy()
```